Documentation/Evolution/UnicodeForStringProcessing.md

```diff
@@ -0,0 +1,779 @@
+
+
+#### POSIX character classes: `[:NAME:]`
+
+**POSIX character classes** represent concepts that we'd like to define at all semantic levels. We propose the following definitions, some of which have been described above. When matching with grapheme cluster semantics, Unicode properties are extended to `Character`s as descrived in the rationale above, and as shown in the table below. That is, for POSIX class `[:word:]`, any `Character` that starts with a matching scalar is a match, while for `[:digit:]`, a matching `Character` must only comprise a single Unicode scalar value.
+
+| POSIX class  | Unicode property class            | Character behavior   | ASCII mode value              |
+|--------------|-----------------------------------|----------------------|-------------------------------|
+| `[:lower:]`  | `\p{Lowercase}`                   | starts-with          | `[a-z]`                       |
+| `[:upper:]`  | `\p{Uppercase}`                   | starts-with          | `[A-Z]`                       |
+| `[:alpha:]`  | `\p{Alphabetic}`                  | starts-with          | `[A-Za-z]`                    |
+| `[:alnum:]`  | `[\p{Alphabetic}\p{Decimal}]`     | starts-with          | `[A-Za-z0-9]`                 |
+| `[:word:]`   | See \* below                      | starts-with          | `[[:alnum:]_]`                |
+| `[:digit:]`  | `\p{DecimalNumber}`               | single-scalar        | `[0-9]`                       |
+| `[:xdigit:]` | `\p{Hex_Digit}`                   | single-scalar        | `[0-9A-Fa-f]`                 |
+| `[:punct:]`  | `\p{Punctuation}`                 | starts-with          | `[-!"#%&'()*,./:;?@[\\\]{}]`  |
+| `[:blank:]`  | `[\p{Space_Separator}\u{09}]`     | starts-with          | `[ \t]`                       |
+| `[:space:]`  | `\p{Whitespace}`                  | starts-with          | `[ \t\n\r\f\v]`               |
+| `[:cntrl:]`  | `\p{Control}`                     | starts-with          | `[\x00-\x1f\x7f]`             |
+| `[:graph:]`  | See \*\* below                    | starts-with          | `[^ [:cntrl:]]`               |
+| `[:print:]`  | `[[:graph:][:blank:]--[:cntrl:]]` | starts-with          | `[[:graph:] ]`                |
+
+\* The Unicode scalar property definition for `[:word:]` is `[\p{Alphanumeric}\p{Mark}\p{Join_Control}\p{Connector_Punctuation}]`.  
+\*\* The Unicode scalar property definition for `[:cntrl:]` is `[^\p{Space}\p{Control}\p{Surrogate}\p{Unassigned}]`.
+
+#### Custom classes
+
+Custom classes function as the set union of their individual components, whether those parts are individual characters, individual Unicode scalar values, ranges, Unicode property classes or POSIX classes, or other custom classes.
+
+- Individual characters and scalars will be tested using the same behavior as if they were listed in an alternation. That is, a custom character class like `[abc]` is equivalent to `(a|b|c)` under the same options and modes.
```

Comment:
I thought we said this was explicitly _not_ the case, so that you could support `e[\u{301}-\u{305}]` or some such, under the impression that alternation required a grapheme break. Has that changed?

Reply:
That's a good point, the `(a|b|c)` grouping would require a grapheme break, which shouldn't be the case. I'll update this language.

Reply:
BTW, what's the reason alternation requires a grapheme break again?

Reply:
Alternation start and end points are set by grouping (or the start/end of the whole pattern), which are the places we're requiring a grapheme break.

Reply:
We surely need them for capturing groups and for the overall match, but even for non-capturing groups? The latter is just syntactic scopes. In the builder that's be an embedded `ChoiceOf { ... }`